pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -94,7 +94,7 @@ def __init__(
         amp_type: str = "native",
         amp_level: Optional[str] = None,
         sync_batchnorm: bool = False,
-        benchmark: bool = False,
+        benchmark: Optional[bool] = None,
```

Comment:
quick ques: why not deprecate this? we can keep this for next 2 release cycles and then remove it.

Reply:
Thanks, @rohitgr7 for the question.

If you meant deprecating the `benchmark` parameter, the reason is that we want to allow users to overwrite this flag manually, having a `benchmark` flag is quite intuitive in my opinion, and must be used a lot. Deprecation of this parameter might be a good discussion for a new issue if you think it's a valid direction to think about.

Reply:
I was just checking the issue, and that's what is proposed there. do you know what benefits do we have when both of these values are the same?

Reply:
Please see a related comment by @carmocca here: https://github.com/PyTorchLightning/pytorch-lightning/pull/11944/files#r808365722 on `benchmark=True, deterministic=True`. I am not sure about the benefits, I think if the user specifies the flags manually, those should be used. cc-ing: @carmocca who might be able to add something here.

Regarding deprecating this flag, I think `torch.backends.cudnn.deterministic` and `torch.backends.cudnn.benchmark` flags do two different things, and since PyTorch allows you to set such flags, it's probably a good idea to have them in the `Trainer` constructor. I'm not sure if PyTorch raises a warning (it doesn't on my machine), but lots of discussions happened [here](https://github.com/pytorch/pytorch/issues/6351).

Reply:
as discussed on the call let's not deprecate it and add a comment so we can look at this in future.

Reply:
A warning has been added + refined test.